app/controllers/templates_controller.rb

```diff
@@ -79,13 +79,17 @@ def auto_complete_controller_name
   end
 
   def preview
+    if params[:id]
```

Comment:
There is a problem returning the before_filter :find_resource because it returns an error page in case it doesn't find a resource, its either rewriting :find_resource(which i rather not touch) or omitting it from the before_filter and using it just when needed(as done here).